erpnext/stock/doctype/pick_list/pick_list.py

```diff
@@ -187,6 +187,8 @@ def set_item_locations(self, save=False):
 		from_warehouses = None
 		if self.parent_warehouse:
 			from_warehouses = get_descendants_of("Warehouse", self.parent_warehouse)
+			if not from_warehouses:
```

Comment:
@tinomhishi I think you should just replace the from warehouses line in 189 with your change instead of conditionally applying it as you have here.